### PR TITLE
[TM-5289] Position Classifications Bug

### PR DESCRIPTION
**Pull request by jlam-mpc**
Bugs:
- Multiple cards denote loading state when only one card is trying to load
- Checkbox form on all cards sync up when they should work independently of one another
- User is able to make draft edits on multiple position classifications and publishable positions at once
- No real data reload after position classifications edit save
- Styling issues
- Using Redux to fetch and capture position classifications for each card in a list doesn't work logically (the state will keep updating to be the new position's data for each card instead of withholding the prior ones)

Changes:
- User can only edit either position details or classifications for ONE card at a time
  - Added edit toggle for position classifications (on initial load everything is disabled and read only)
  - If a classification tab is in edit mode, the same logic for editing details applies and user cannot update search or edit other cards
- Fixed line separated fields styling on all components that use it
- Converted from Redux to useDataLoader to load each card's classifications independently
  - useDataLoader hook has new refetch parameter to manually control when to refetch (helps with reloading after successful edit)
  - Also resolves independent loading states between each card (had to toss old logic from Charles- sorry)
- Fixed many console warnings

[TM-5289](https://metaphase.atlassian.net/browse/TM-5289)

[TM-5289]: https://metaphase.atlassian.net/browse/TM-5289?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Components/PublishablePositionCard/PositionClassification/PositionClassification.jsx [hunk 1]
```diff
@@ -1,34 +1,44 @@
...
+    undefined,
+    refetch,
+  );
+
+
   useEffect(() => {
-    setClassifications(pc);
-    setSelections(cs);
+    setClassifications(results?.data?.positionClassifications ?? []);
+    setSelections(results?.data?.classificationSelections ?? []);
```
Comment (scott062) [inline src/Components/PublishablePositionCard/PositionClassification/PositionClassification.jsx:37]: || vs ??

I assume we want it on any falsey value?

File: src/Components/PublishablePositionCard/PositionClassification/PositionClassification.jsx [hunk 2]
```diff
@@ -40,14 +50,20 @@ const PositionClassification = (props) => {
     setSelections(newSelections);
   };
 
+  const handleCancel = () => {
+    setEditMode(false);
+    setSelections(results?.data?.classificationSelections ?? []);
```
Comment (scott062) [inline src/Components/PublishablePositionCard/PositionClassification/PositionClassification.jsx:55]: See above

File: src/Components/PublishablePositionCard/PositionClassification/PositionClassification.jsx [hunk 3]
```diff
@@ -40,14 +50,20 @@ const PositionClassification = (props) => {
...
+
   const handleSubmit = () => {
     let position = '';
     let codes = '';
     let values = '';
     let updatedDates = '';
     let updaterIds = '';
 
-    selections.forEach(s => {
+    /* eslint-disable no-unused-expressions */
```
Comment (scott062) [inline src/Components/PublishablePositionCard/PositionClassification/PositionClassification.jsx:65]: Not sure of the eslint error here - would it help to put this code in a returnable formatInput()? If it works then no worries

File: src/Components/PublishablePositionCard/PositionClassification/PositionClassification.jsx [hunk 4]
```diff
@@ -96,32 +141,42 @@ const PositionClassification = (props) => {
                   <input
                     type="checkbox"
                     name={o.code}
-                    checked={selections.find(s => o.code === s.code && s.value === '1') ?? false}
+                    checked={selections?.find(s => o.code === s.code && s.value === '1') ?? false}
```
Comment (scott062) [inline src/Components/PublishablePositionCard/PositionClassification/PositionClassification.jsx:144]: See above

File: src/Components/PublishablePositionCard/PositionClassification/PositionClassification.jsx [hunk 5]
```diff
@@ -96,32 +141,42 @@ const PositionClassification = (props) => {
...
+      }
     </div>
   );
 };
 
 PositionClassification.propTypes = {
   positionNumber: PropTypes.string.isRequired,
   bureau: PropTypes.string.isRequired,
-  posSeqNum: PropTypes.string.isRequired,
+  posSeqNum: PropTypes.number.isRequired,
```
Comment (scott062) [inline src/Components/PublishablePositionCard/PositionClassification/PositionClassification.jsx:167]: Good catch - I think we have a proptype to handle either scenario if its not consistent. If it is always number then even better

File: src/Components/PublishablePositionCard/PositionClassification/PositionClassification.jsx [hunk 6]
```diff
@@ -56,14 +72,32 @@ const PositionClassification = (props) => {
...
-        updater_ids: updaterIds,
-        updated_dates: updatedDates,
-      }));
+      api().put('/fsbid/position_classifications/edit/', editData)
+        .then(() => {
+          const toastTitle = UPDATE_POSITION_CLASSIFICATION_SUCCESS_TITLE;
+          const toastMessage = UPDATE_POSITION_CLASSIFICATION_SUCCESS;
+          batch(() => {
+            dispatch(toastSuccess(toastMessage, toastTitle));
+            setRefetch(!refetch);
```
[the remark this answers is not in the document]
Comment (scott062) [reply src/Components/PublishablePositionCard/PositionClassification/PositionClassification.jsx:90]: jk - i see the implementation in the hook relies on this changing

File: src/Components/BureauPage/PositionManager/PositionManagerSearch/PositionManagerSearch.jsx [hunk 7]
```diff
@@ -50,6 +50,7 @@ const PositionManagerSearch = forwardRef((props, ref) => {
         noButton={props.noButton}
         showIcon={props.showIcon}
         inputDisabled={props.disableSearch}
+        noForm
```
Comment (scott062) [inline src/Components/BureauPage/PositionManager/PositionManagerSearch/PositionManagerSearch.jsx:53]: Is this just b/c we already wrapped the search bar in another form?

File: src/Components/PublishablePositionCard/PositionClassification/PositionClassification.jsx (hunk 1 above)
Comment (jlam-mpc) [reply src/Components/PublishablePositionCard/PositionClassification/PositionClassification.jsx:37]: Apologies but yes- just fixed

File: src/Components/BureauPage/PositionManager/PositionManagerSearch/PositionManagerSearch.jsx (hunk 7 above)
Comment (jlam-mpc) [reply src/Components/BureauPage/PositionManager/PositionManagerSearch/PositionManagerSearch.jsx:53]: Yeah so I could either remove the form wrapping this or use the noForm prop which already exists, either way works. This just resolves one of the many many console warnings on this page.

**Review by scott062 — APPROVED** [review]

**Pull request merged**